spec/factories/categories.rb

```diff
@@ -0,0 +1,10 @@
+FactoryBot.define do
+  factory :category do
+    name { Faker::Commerce.department }
+    parent_category_id { nil }
```

Comment:
Here we can use the association directly, not the db column https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#implicit-definition
```suggestion
    association :parent_category, strategy: :null
```

The benefit of this is that you can send the instance of the category directly, without asking for the id